staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -457,3 +459,58 @@ func ParseApplySetParentRef(parentRefStr string, mapper meta.RESTMapper) (*Apply
 	}
 	return &ApplySetParentRef{Name: name, RESTMapping: mapping}, nil
 }
+
+// Prune deletes any objects from the apiserver that are no longer in the applyset.
+func (a *ApplySet) Prune(ctx context.Context, o *ApplyOptions) error {
```

Comment:
Same question as below about having a `Prune` and a `pruneAll` on the same object (though this one does more than just wrap). Also, now that applySetPruner isn't a separate object, should that file's content move here? Or if we're collecting the pruning related functionality in that file, shouldn't this and the method below be there?

Reply:
Separate package in a later refactor SGTM!